src/solvers/flattening/boolbv_add_sub.cpp

```diff
@@ -41,8 +39,9 @@ bvt boolbvt::convert_add_sub(const exprt &expr)
 
   if(op0.type()!=type)
   {
-    std::cerr << expr.pretty() << '\n';
-    throw "add/sub with mixed types";
+    error().source_location = expr.find_source_location();
+    error() << "add/sub with mixed types:\n" << expr.pretty() << eom;
+    throw 0;
```

Comment:
:-\  Please could we not increase the number of throw 0's.  This seems to me a pretty clear:

DATA_INVARIANT(op0.type()!=type, "add/sub arguments must have the same type")